### PR TITLE
Updates for compatibility with ActionHero v18

### PR DESCRIPTION
**Pull request by bayssmekanique**
Uses ECMAScript 2017 features sets
- `async/await` instead of callbacks
- `ActionHero` class extension

Updates documentation to reflect Sequelize `4.x` changes and ActionHero `18.x` changes.

Peer dependencies added to `package.json`.  Allows for Sequelize versions `3.x` or `4.x`, but requires ActionHero `+18`.

_I'm not certain on initializer priorities.  The previous version was set to line up with `redis`, but those values seemed too extreme for this plugin.  Please advise!_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,35 +4,69 @@
...
+    paranoid: true
   })
+
+  // Attach Class methods
+  model.rehydrate = function (user) {
+    return this.build(user)
+  }
+
+  // Attach Instance methods
+  model.prototype.apiData = function () {
```
Comment (evantahler) [inline README.md]: 👍

File: README.md [hunk 2]
```diff
@@ -42,47 +76,41 @@ An example migration to create a `users` table would look like:
...
+      name: DataTypes.STRING,
+      email: DataTypes.STRING,
+      phone: DataTypes.STRING,
+      passwordHash: DataTypes.TEXT,
+      passwordSalt: DataTypes.TEXT,
+      createdAt: DataTypes.DATE,
+      updatedAt: DataTypes.DATE
+    })
+
+    await migration.addIndex('users', ['email'], {
```
Comment (evantahler) [inline README.md]: This syntax is so much better!

File: initializers/sequelize.js [hunk 3]
```diff
@@ -1,167 +1,24 @@
...
+    constructor () {
+      super()
+      this.name = 'sequelize'
+      this.loadPriority = 101
+      this.startPriority = 101
+      this.stopPriority = 300
+    }
+
+    async initialize () {
+      ActionHero.api.models = {}
```
Comment (evantahler) [inline initializers/sequelize.js]: It's semantic, but the convention would be more like:
```js
const {api} = require('actionhero')
//...
api.models = {}
api.sequelize = new SequelizePlugin()

```

File: scripts/postinstall.js [hunk 4]
```diff
@@ -1,38 +1,37 @@
...
+const localRcFile = path.normalize(path.join(__dirname, '/../config/.sequelizerc'))
+const projectRcFile = path.normalize(process.cwd() + '/../../.sequelizerc')
+
+try {
+  fs.lstatSync(projectConfigFile)
+} catch (ex) {
+  // unable to stat file because it doesn't exist
+  console.log('copying ' + localConfigFile + ' to ' + projectConfigFile)
+  mkdirp.sync(path.normalize(projectConfigDir))
+  fs.createReadStream(localConfigFile).pipe(fs.createWriteStream(projectConfigFile))
```
Comment (evantahler) [inline scripts/postinstall.js]: This isn't needed any more, as linking a plugin copies config files automatically: https://github.com/actionhero/actionhero/blob/master/bin/methods/link.js#L91-L92

**Comment by evantahler** [comment]
@bayssmekanique the reason the initializer priority is so low, is that other initializers may rely on this already working.  I think since it is self-contained, that is OK.

**Comment by bayssmekanique** [comment]
The reason for extracting the `Sequelize` initialization logic into its own class was largely for encapsulation.  It seemed more logical that all the properties of the `api.sequelize` object would be contained in a single class definition, and it might make it easier to implement typed definitions sometime down the road.

**Pull request merged**